zend/callable.cpp

```diff
@@ -147,10 +152,12 @@ void Callable::initialize(zend_internal_function_info *info, const char *classna
 #else
     // the properties that are available on php 7.2 and higher
     info->required_num_args = _required;
+#if PHP_VERSION_ID < 80000
     info->return_reference = false;
     info->_is_variadic = false;
     info->type = ZEND_TYPE_ENCODE((int)_return, true);
```

Comment:
I wonder if there needs to be a similar line for PHP 8 or not ?

Reply:
I was running into various SIGSEV's with functions that utilised pass by reference.

```
__strchrnul (s=0x202c6e6f69746169 <error: Cannot access memory at address 0x202c6e6f69746169>, c=c@entry=124) at src/string/strchrnul.c:19
19	src/string/strchrnul.c: No such file or directory.
(gdb) backtrace
#0  __strchrnul (s=0x202c6e6f69746169 <error: Cannot access memory at address 0x202c6e6f69746169>, c=c@entry=124) at src/string/strchrnul.c:19
#1  0x00007fee50a5f4f4 in strchr (s=<optimized out>, c=124) at src/string/strchr.c:5
#2  0x000055d31751a242 in zend_register_functions ()
#3  0x000055d31751a7aa in ?? ()
#4  0x00007fee4fed96e5 in Php::ClassImpl::initialize (this=0x7fee5020b7f0, base=0x7fee4f93bcd0, prefix=...) at zend/classimpl.cpp:1457
```

The following fixed it:

```
#if PHP_VERSION_ID >= 80000
    info->type = (zend_type) ZEND_TYPE_INIT_CODE((int)_return, true, 0);
#endif
```

Reply:
Thanks!, I've added it to my PR/patch.